src/textadept.c

```diff
@@ -71,6 +84,10 @@ bool emit(const char *name, ...) {
 			ref = va_arg(ap, int);
 			lua_rawgeti(lua, LUA_REGISTRYINDEX, ref), luaL_unref(lua, LUA_REGISTRYINDEX, ref);
 			break;
+		case LUA_TBUFFER:
+			lua_pushdoc(lua, va_arg(ap, sptr_t)); break;
+		case LUA_TVIEW:
+			lua_pushview(lua, va_arg(ap, SciObject *)); break;
```

Comment:
No, sorry, I don't like this. Is there something preventing us from the usual `emit("...", LUA_TTABLE, (lua_pushdoc(L, ...), luaL_ref(L, LUA_REGISTRYINDEX)), -1)`?